### ISSUE TITLE
Live event called too early

### ISSUE THREAD

**Issue by oiledCode** [jhuckaby/webcamjs#84]
First of all I would like to thank you for making your code open source, is it helping me a lot. 
I've noticed anyway that the live event is called too early, and this lead to a weird visual effect, first I see a black area and after one sec (I did not measure) the preview is visible. So I had a look at the code and I noticed this :  line 123 of webcam.js
```javascript
function(stream) {
   // got access, attach stream to video
   video.src = window.URL.createObjectURL( stream ) || stream;
   Webcam.stream = stream;
   Webcam.loaded = true;
   Webcam.live = true;
   Webcam.dispatch('load');
   Webcam.dispatch('live');
   Webcam.flip();
},
```
so basically as soons as the success callback of getUserMedia is called you dispatch the live event. 
Now if I've understood the meaning of your live event (camera preview visible) you should dispatch the live event once the video element has loaded the metadata. Something like this for example :

```javascript
function(stream) {
   video.onloadedmetadata = function(e) {
      Webcam.stream = stream;
      Webcam.loaded = true;
      Webcam.live = true;
      Webcam.dispatch('load');
      Webcam.dispatch('live');
      Webcam.flip();
   }
   video.src = window.URL.createObjectURL( stream ) || stream;
},
```
This in theory should guarantee that the live event is called when the preview is actually visible. 
What do you think?

Thanks for your time

**Comment by escobar5** [comment]
I'm having the same problem, this fix works for me. Thanks

**Comment by gintechsystems** [comment]
This fix also worked for me, should definitely be updated.

**Comment by jhuckaby** [comment]
Thanks everyone!  I will definitely make this change in the script.

**Comment by jhuckaby** [comment]
Fixed in v1.0.9, thanks!
